src/main/java/io/candyjar/controller/CandyJarRestController.java

```diff
@@ -36,6 +36,7 @@ public Candy addCandy(@RequestBody Candy candy) {
 
 	@PutMapping("/candy/{id}")
 	public Candy updateCandy(@PathVariable Long id, @RequestBody Candy candy) {
+		candy.setId(id);
```

Comment:
We shouldn't need to set the Id on the candy object since it's being passed in with the Path and we're selecting from the database to get that object back.